pkg/config/config_linux.go

```diff
@@ -15,9 +15,7 @@ func isContainer() bool {
 
 	if strings.Contains(string(data), "docker") ||
 		strings.Contains(string(data), "/lxc/") ||
-		[]string{string(data)}[0] != "systemd" &&
-			[]string{string(data)}[0] != "init" ||
-		os.Getenv("container") != "" {
+		os.Getenv("CONTAINER") != "" {
```

Comment:
Your change is certainly an improvement, but I find this condition still strange for two reasons: it checks the error only after looking for the strings (which is probably fine in practice, as `data` is likely to be empty when there was an error, but it's still weird), and then the funny `&& true` at the end.

I'd clean it up a bit further to something like
```go
	return err == nil && (strings.Contains(string(data), "docker") ||
		strings.Contains(string(data), "/lxc/") ||
		os.Getenv("CONTAINER") != "")
```